pilot/pkg/config/kube/gateway/conversion.go

```diff
@@ -1939,13 +1964,44 @@ type routeParentReference struct {
 	// OriginalReference contains the original reference
 	OriginalReference k8s.ParentReference
 	// Hostname is the hostname match of the parent, if any
-	Hostname string
+	Hostname        string
+	BannedHostnames sets.Set[string]
 }
 
 func (r routeParentReference) IsMesh() bool {
 	return r.InternalName == "mesh"
 }
 
+func (r routeParentReference) hostnameAllowedByIsolation(rawRouteHost string) bool {
+	routeHost := host.Name(rawRouteHost)
+	ourListener := host.Name(r.Hostname)
+	if len(ourListener) > 0 && !ourListener.IsWildCarded() {
+		// Short circuit: this logic only applies to wildcards
+		// Not required for correctness, just an optimization
+		return true
+	}
+	if len(ourListener) > 0 && !routeHost.Matches(ourListener) {
+		return false
+	}
+	for checkListener := range r.BannedHostnames {
+		// We have 3 hostnames here:
+		// * routeHost, the hostname in the route entry
+		// * ourListener, the hostname of the listener the route is bound to
+		// * checkListener, the hostname of the other listener we are comparing to
+		// We want to return false if checkListener would match the routeHost and it would be a more exact match
+		if len(ourListener) > len(checkListener) {
+			// If our hostname is longer, it must be more exact than the check
+			continue
+		}
+		// Ours is shorter. If it matches the checkListener, then it should ONLY match that one
+		// Note protocol, port, etc are already considered
```

Comment:
where are they considered?

Reply:
In appendParent when we compute bannedHostnames. Ill update the comment